books/free-programming-books-subjects.md

```diff
@@ -733,7 +733,9 @@ Books that cover a specific programming language can be found in the [BY PROGRAM
 * [Is Parallel Programming Hard, And, If So, What Can You Do About It?](https://www.kernel.org/pub/linux/kernel/people/paulmck/perfbook/perfbook.html) - Paul E. McKenney
 * [Programming on Parallel Machines; GPU, Multicore, Clusters and More](http://heather.cs.ucdavis.edu/parprocbook) - Norm Matloff
 Kerridge (PDF) (email address *requested*, not required)
-* [The OpenCL Programming Book](https://us.fixstars.com/products/opencl/book/OpenCLProgrammingBook/contents/)
+* [The OpenCL Programming Book](https://web.archive.org/web/20210321083327/https://us.fixstars.com/products/opencl/book/OpenCLProgrammingBook/contents/) *(:card_file_box: archived)*
+
+
```

Comment:
Remove lines 737 and 738 for fixing the blank line issue, and remember to check for linter errors like alphabetical order and spacing when contributing in the future.